tests/integration_test.go

```diff
@@ -1,82 +1,129 @@
 package integration_test
 
 import (
+	"bytes"
 	"context"
 	"io/ioutil"
+	"log"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"testing"
 
 	"github.com/numtide/nar-serve/go-nix/libstore"
+	"github.com/stretchr/testify/assert"
 )
 
+func cmd(env []string, name string, args ...string) *exec.Cmd {
+	cmd := exec.Command(name, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Env = env
+
+	return cmd
+}
+
 func TestHappyPath(t *testing.T) {
+	assert := assert.New(t)
 	accessKeyID := "Q3AM3UQ867SPQQA43P2F"
 	secretAccessKey := "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG"
+	parentDir := os.TempDir()
+	env := append(os.Environ(),
+		"AWS_ACCESS_KEY_ID="+accessKeyID,
+		"AWS_SECRET_ACCESS_KEY="+secretAccessKey,
+		"MINIO_ACCESS_KEY="+accessKeyID,
+		"MINIO_SECRET_KEY="+secretAccessKey,
+		"MINIO_REGION_NAME=us-east-1",
+		"HOME="+parentDir,
+	)
 
-	dataDir, err := ioutil.TempDir("", "nar")
+	dataDir, err := ioutil.TempDir("", "nar-serve")
 	if err != nil {
-		panic(err)
+		t.Fatal("tmpdir error:", err)
 	}
 	defer os.RemoveAll(dataDir)
+	log.Println("datadir", dataDir)
 
-	minios := exec.Command("minio", "server", dataDir)
-	minios.Env = []string{
-		"MINIO_ACCESS_KEY=" + accessKeyID,
-		"MINIO_SECRET_KEY=" + secretAccessKey,
-	}
+	// Start the server
+	minios := cmd(env, "minio", "server", dataDir, "--config-dir", ".")
```

Comment:
```suggestion
	minios := cmd(env, "minio", "server", dataDir, "--config-dir", configDir)
```
point each of these to unique folders that get deleted after the test run